torch/fx/experimental/symbolic_shapes.py

```diff
@@ -3533,7 +3538,11 @@ def size_hint(self, expr: "sympy.Expr"):
             r = self._maybe_evaluate_static(result_expr, compute_hint=True)
             if r is not None:
                 return r
-            raise self._make_data_dependent_error(result_expr, expr)
+            # if `config.capture_scalar_outputs`` is true, allow unbacked symint in size hint
+            if torch._dynamo.config.capture_scalar_outputs:
```

Comment:
This is probably the wrong config to control this, and I'm also not convinced we shouldn't already have slammed in the heuristic value at this point